backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt

```diff
@@ -104,4 +104,103 @@ class ShorkInterpreterControllerV1IT() : AbstractControllerTest() {
     }
+
+    @Test
+    fun `test player code submission with valid username and its retrieval`() = runTest {
+        val player = "playerA"
+        client.post("/api/v1/lobby/0/code/$player") {
+            contentType(ContentType.Application.Json)
+            setBody("someString")
+        }
+        val result = client.get("/api/v1/lobby/0/code/playerA")
+        assertEquals(
+            "someString",
+            Json.parseToJsonElement(result.bodyAsText()).jsonObject["code"]?.jsonPrimitive?.content,
+        )
+    }
+
+    @Test
+    fun `test player empty code submission`() = runTest {
+        val player = "playerA"
+        val result = client.post("/api/v1/lobby/0/code/$player")
+        assertEquals(HttpStatusCode.OK, result.status)
+    }
+
+    @Test
+    fun `test player code submission with invalid username does not change lobby status`() =
+        runTest {
+            val player = "playerC"
+            client.post("/api/v1/lobby/0/code/$player") {
+                contentType(ContentType.Application.Json)
+                setBody("someString")
+            }
+
+            val result = client.get("/api/v1/lobby/0/code/playerC")
+            assertEquals(HttpStatusCode.BadRequest, result.status)
+            assert(result.bodyAsText().contains("No player with that name in the lobby"))
+
+            val responseData = parseStatus(result)
+            assertEquals("false", responseData["playerASubmitted"])
+            assertEquals("false", responseData["playerBSubmitted"])
+            assertEquals("NOT_STARTED", responseData["gameState"])
+            assertEquals("UNDECIDED", responseData["result.winner"])
+        }
+
+    @Test
+    fun `test player code submission reflecting in lobby status`() = runTest {
+        val player = "playerA"
+        client.post("/api/v1/lobby/0/code/$player") {
+            contentType(ContentType.Application.Json)
+            setBody("someString")
+        }
+
+        val result = client.get("/api/v1/lobby/status/0")
+        val responseData = parseStatus(result)
+        assertEquals("true", responseData["playerASubmitted"])
+        assertEquals("false", responseData["playerBSubmitted"])
+        assertEquals("NOT_STARTED", responseData["gameState"])
+        assertEquals("UNDECIDED", responseData["result.winner"])
+    }
+
+    @Test
+    fun `test game starts when both players submit and longer code wins`() = runTest {
+        client.post("/api/v1/lobby/0/code/playerA") {
+            contentType(ContentType.Application.Json)
+            setBody("someVeryLongString")
+        }
+
+        client.post("/api/v1/lobby/0/code/playerB") {
+            contentType(ContentType.Application.Json)
+            setBody("someShortString")
+        }
+
+        val result = client.get("/api/v1/lobby/status/0")
+        val responseData = parseStatus(result)
+        assertEquals("FINISHED", responseData["gameState"])
+        assertEquals("A", responseData["result.winner"])
+    }
```

Comment:
I can see why testing that uploading for both players works could be beneficial, but asserting that playerA is the winner just tests the MockShork not the upload endpoints.
I'd prefer the test without the last line and an appropriate rename to something like `test game starts when both players submit`